File: backend/app/schemas/risk.py
from pydantic import BaseModel, Field


class PreTradeRiskRequest(BaseModel):
    symbol: str = "BTC-USD"
    side: str = "long"
    size_usd: float = Field(gt=0)
    entry_price: float | None = None


class RiskContributor(BaseModel):
    factor: str
    impact: float
    description: str


class PreTradeRiskResponse(BaseModel):
    similarity_to_losing_setups: float
    severity: str
    title: str
    summary: str
    contributors: list[RiskContributor]
    recommended_action: str
    confidence: float

File: frontend/src/lib/api/risk.ts
import { api } from '@/lib/api/client'

export interface RiskContributor {
  factor: string
  impact: number
  description: string
}

export interface PreTradeRiskResponse {
  similarity_to_losing_setups: number
  severity: string
  title: string
  summary: string
  contributors: RiskContributor[]
  recommended_action: string
  confidence: number
}

export async function evaluatePreTradeRisk(symbol = 'BTC-USD', side = 'long', size_usd = 5000): Promise<PreTradeRiskResponse> {
  return api.post('/api/v1/risk/pretrade', { symbol, side, size_usd })
}
